packages/workspace/src/node/default-workspace-server.ts

```diff
@@ -109,11 +116,25 @@ export class DefaultWorkspaceServer implements WorkspaceServer {
     private async readFromUserHome(): Promise<WorkspaceData | undefined> {
         const file = this.getUserStoragePath();
         if (await fs.pathExists(file)) {
-            const config = await fs.readJson(file);
+            const rawContent = await fs.readFile(file, 'utf-8');
+            const content = rawContent.trim();
+            if (!content) {
```

Comment:
I don't think this works if the intent is to catch an empty file. Maybe instead test if the length of content is equal to zero?